### PR TITLE
LBaaS v2 l7 policy support - part 4: delete l7policy

### PR DESCRIPTION
**Pull request by lingxiankong**
For #832

Neutron-LBaaS l7 policy delete API implementation:
https://github.com/openstack/neutron-lbaas/blob/8e8a6c47c7d38fa7b61850ff9ea2cf130718ded3/neutron_lbaas/services/loadbalancer/plugin.py#L954

Octavia l7 policy delete API implementation:
https://github.com/openstack/octavia/blob/54a4cf00cf304b108013ca4487c138c13f30983d/octavia/api/v2/controllers/l7policy.py#L253

### CODE REVIEW

**Comment by lingxiankong** [comment]
@jrperritt @jtopjian this patch is also ready for review

File: acceptance/openstack/loadbalancer/v2/loadbalancers_test.go [hunk 1]
```diff
@@ -101,10 +102,18 @@ func TestLoadbalancersCRUD(t *testing.T) {
 	tools.PrintResource(t, newListener)
 
 	// L7 policy
-	_, err = CreateL7Policy(t, lbClient, listener, lb)
+	policy, err := CreateL7Policy(t, lbClient, listener, lb)
 	if err != nil {
 		t.Fatalf("Unable to create l7 policy: %v", err)
 	}
+	defer DeleteL7Policy(t, lbClient, policy.ID)
```
Comment (jtopjian) [inline acceptance/openstack/loadbalancer/v2/loadbalancers_test.go]: This should be

```go
defer DeleteL7Policy(t, lbClient, lb.ID, policy.ID)
```

Comment (lingxiankong) [reply acceptance/openstack/loadbalancer/v2/loadbalancers_test.go]: Yes. I wonder why there is no complaint in the CI?

Comment (jtopjian) [reply acceptance/openstack/loadbalancer/v2/loadbalancers_test.go]: Because OpenLab only runs a subset of tests at this time.

**Comment by jtopjian** [comment]
LGTM - thank you :)

**Comment by mrhillsman** [comment]
recheck

**Pull request merged**